File: sample/MainActivity.cs
using System;
using Android.App;
using Android.OS;
using Android.Runtime;
using Android.Views;
using AndroidX.AppCompat.Widget;
using AndroidX.AppCompat.App;
using Google.Android.Material.FloatingActionButton;
using Com.Zendesk.Logger;
using Com.Zendesk.Service;
using GoogleGson.Stream;
using Zendesk.Core;
using Zendesk.Support;
using Zendesk.Support.Guide;
using Zendesk.Support.Request;
using Zendesk.Support.Requestlist;
using Zendesk.Chat;
using Zendesk.Messaging;

namespace SampleApp
{
    [Activity(
        Label = "@string/app_name",
        Theme = "@style/AppTheme.NoActionBar",
        MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            Xamarin.Essentials.Platform.Init(this, savedInstanceState);

            SetContentView(Resource.Layout.activity_main);

            var toolbar = FindViewById<Toolbar>(Resource.Id.toolbar);
            SetSupportActionBar(toolbar);

            var fab = FindViewById<FloatingActionButton>(Resource.Id.fab);
            fab.Click += FabOnClick;


            InitZendesk();
        }

        private void InitZendesk()
        {
            // Enable Zendesk SDK logs
            Logger.Loggable = true;

            // TODO: Add valid credentials:
            // Init SDK
            Zendesk.Core.Zendesk.Instance.Init(this,
                zendeskUrl: "",
                applicationId: "",
                oauthClientId: "");

            // Init user
            var identity = new AnonymousIdentity.Builder()
                .WithNameIdentifier(name: "TestUserName")
                .WithEmailIdentifier(email: "test@test.com")
                .Build();
            Zendesk.Core.Zendesk.Instance.Identity = identity;

            // Init Support
            // https://developer.zendesk.com/documentation/classic-web-widget-sdks/support-sdk/android/nutshell/
            Support.Instance.Init(Zendesk.Core.Zendesk.Instance);

            // Init Chat
            // https://developer.zendesk.com/documentation/classic-web-widget-sdks/chat-sdk-v2/android/getting-started/
            // https://developer.zendesk.com/documentation/classic-web-widget-sdks/chat-sdk-v2/android/quick-start-chat-sdk-android/
            Chat.Instance.Init(
                    this,
                    "Your accountKey",
                    "Your appId");
        }

        private void FabOnClick(object sender, EventArgs eventArgs)
        {
            // Simple configuration for chat
            ChatConfiguration chatConfiguration = ChatConfiguration.InvokeBuilder()
                .WithAgentAvailabilityEnabled(false)
                .Build();

            // Show Zendesk Chat
            MessagingActivity.Builder()
                .WithEngines(ChatEngine.Engine())
                .Show(this, chatConfiguration);
        }

        public override bool OnCreateOptionsMenu(IMenu menu)
        {
            MenuInflater.Inflate(Resource.Menu.menu_main, menu);
            return true;
        }

        public override bool OnOptionsItemSelected(IMenuItem item)
        {
            switch (item.ItemId)
            {
                case Resource.Id.action_help:
                    HelpCenterActivity.Builder().Show(this);
                    return true;

                case Resource.Id.action_tickets:
                    RequestListActivity.Builder().Show(this);
                    return true;

                case Resource.Id.action_request:
                    RequestActivity.Builder().Show(this);
                    return true;

                default:
                    return base.OnOptionsItemSelected(item);
            }
        }

        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
        {
            Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);

            base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        }
    }
}

File: tools/TestZendeskNative/app/src/main/java/com/example/testzendesknative/MainActivity.java
package com.example.testzendesknative;

import static zendesk.support.HelpCenterTracker.*;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.zendesk.logger.Logger;

import zendesk.chat.Chat;
import zendesk.chat.ChatConfiguration;
import zendesk.chat.ChatEngine;
import zendesk.core.AnonymousIdentity;
import zendesk.core.Identity;
import zendesk.core.Zendesk;
import zendesk.messaging.MessagingActivity;
import zendesk.support.Support;
import zendesk.support.request.RequestActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(view -> {
            //Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
            //        .setAction("Action", null).show();

            // Simple configuration for chat
            ChatConfiguration chatConfiguration = ChatConfiguration.builder()
                    .withAgentAvailabilityEnabled(false)
                    .build();

            // Show Zendesk Chat
            MessagingActivity.builder()
                    .withEngines(ChatEngine.engine())
                    .show(view.getContext(), chatConfiguration);
        });

        // Enable Zendesk SDK logs
        Logger.setLoggable(true);

        // Init SDK
        Zendesk.INSTANCE.init(
                this,
                "Your zendeskUrl",
                "Your appId",
                "Your clientId");

        // Init user
        Identity identity = new AnonymousIdentity();
        Zendesk.INSTANCE.setIdentity(identity);

        // Init Support
        // https://developer.zendesk.com/documentation/classic-web-widget-sdks/support-sdk/android/nutshell/
        Support.INSTANCE.init(Zendesk.INSTANCE);

        // Init Chat
        // https://developer.zendesk.com/documentation/classic-web-widget-sdks/chat-sdk-v2/android/getting-started/
        // https://developer.zendesk.com/documentation/classic-web-widget-sdks/chat-sdk-v2/android/quick-start-chat-sdk-android/
        Chat.INSTANCE.init(
                this,
                "Your accountKey",
                "Your appId");
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_request) {

            // Simple configuration for show new Zendesk Support request
            RequestActivity.builder()
                    .withRequestSubject("Testing Support SDK")
                    .withTags("sdk", "android")
                    .show(this);

            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}